### ISSUE TITLE
🐛 Bug Report: disabled tests for GCP / VertexAI

### ISSUE THREAD

**Issue by nirga** [traceloop/openllmetry#417]
### Which component is this bug for?

VertexAI Instrumentation

### 📜 Description

Following #413, I had to disable the VertexAI tests since vcr.py doesn't support GRPC. We need to figure out how to mock those requests to avoid making actual calls to GRPC.

### 👟 Reproduction steps

N/A

### 👍 Expected behavior

N/A

### 👎 Actual Behavior with Screenshots

N/A

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by MarkWinter21** [comment]
Hi, does this bug still need to be looked at? Alternatively, could you suggest another 'good first issue'. Thanks

**Comment by nirga** [comment]
Yes @MarkWinter21! We can look into either finding something like `vcr.py` that supports GRPC (which is what VertexAi is using), or just create a stub that mocks the calls to VertexAI ourselves.
There's something more interesting for you to look at though, if you want - #1026

**Comment by Nitish-Kshatriya-P** [comment]
@nirga Hey, can I work on this bug if it still exists?

**Comment by PraneethVarma5** [comment]
Hi, I’d like to work on this issue. Please assign it to me.

**Comment by Luciana081** [comment]
Hi! I am working on an AI open source capstone project and would like to work on this issue. I am planning to start by reproducing the disabled VertexAI test problem locally, then investigate how to mock the GRPC-based VertexAI calls so the tests can run without external GCP requests. Please let me know if there is any preferred testing approach I should follow before I begin.

**Comment by Ranga-Prasath-22** [comment]
Hi @nirga, I'd like to work on this issue.

After reviewing the thread and the VertexAI instrumentation code, here's my proposed approach:

**Option 1: Mock at the Python level using unittest.mock** (preferred for simplicity)
- Patch the VertexAI gRPC client methods before test execution
- Create a fixture that intercepts grpc channel calls and returns canned responses
- This avoids needing a new dependency like cr.py for gRPC

**Option 2: Use grpcio-testing** (more robust but heavier)
- Leverage the built-in gRPC test utilities to create in-process test channels

I'd lean toward Option 1 since it's simpler and doesn't add dependencies. I can start by looking at how the existing VertexAI tests are structured in packages/opentelemetry-instrumentation-vertexai/tests/ and then create mock fixtures for the gRPC calls.

Please assign this to me if you'd like me to proceed.

**Comment by rehan243** [comment]
Oh yeah, vcr.py not supporting GRPC is a classic headache. I've dealt with this before, and honestly, the easiest path might be switching to something like `pytest-grpc`. It lets you mock GRPC services directly without needing to intercept actual network calls, which avoids the vcr.py limitation entirely. 

Alternatively, if you really want to stick with vcr.py, you could try wrapping the GRPC calls in HTTP requests with something like a REST proxy and then record those, but that's... kind of hacky and overcomplicates things. 

I'd lean towards setting up mocking directly at the GRPC level, less brittle and no extra dependencies.

**Comment by abdd8433** [comment]
Hi! I’m interested in working on this issue as a first contribution. I plan to investigate the disabled VertexAI tests and add mocked gRPC responses so they can run without making real network calls. Is this issue still available, or is someone currently working on it?
